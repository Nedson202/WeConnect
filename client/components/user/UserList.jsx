import React from 'react';
import PropTypes from 'prop-types';
import UserCard from './UserCard';

/**
   * @description Creates users list.
   * @param {prop} users
   * @param {prop} deleteUser
   * @returns {object} JSX object
   * @memberof UserList
   */
const UserList = ({ users, removeUser, markUserToRemove }) => {
  const noUser = (
    <h4 className="text-center">No user registered yet</h4>
  );

  const userList = (
    <div className="row">
      { users.map(user =>
        (<UserCard
          user={user}
          removeUser={removeUser}
          markUserToRemove={markUserToRemove}
          key={user.id}
        />))}
    </div>
  );

  return (
    <div>
      <div className="text-center">
        {users.length === 0 ? null :
        <h4>Displaying user list</h4>
        }
      </div>
      <div className="container card-container">
        {users.length === 0 ? noUser : userList}
      </div>
    </div>
  );
};

UserList.propTypes = {
  removeUser: PropTypes.func.isRequired,
  markUserToRemove: PropTypes.func.isRequired,
  users: PropTypes.array.isRequired
};

export default UserList;
