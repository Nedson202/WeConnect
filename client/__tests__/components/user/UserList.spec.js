import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import UserList from '../../../components/user/UserList';
import userData from '../../__mockData__/userData';

configure({ adapter: new Adapter() });
const { users } = userData;
let props;
const setup = () => {
  props = {
    users,
    removeUser: jest.fn(() => Promise.resolve()),
    markUserToRemove: jest.fn(() => Promise.resolve())
  };

  return shallow(<UserList {...props} />);
};


describe('Component: UserList', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(4);
    wrapper.setProps({ users: [] });
    expect(wrapper.find('div').length).toBe(3);
    wrapper.setProps({ users });
    expect(wrapper.find('div').length).toBe(4);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h4').length).toBe(1);
    expect(wrapper.length).toBe(1);
  }) 
});