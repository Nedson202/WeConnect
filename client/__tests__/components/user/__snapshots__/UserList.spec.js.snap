// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserList should render 1`] = `
<div>
  <div
    className="text-center"
  >
    <h4>
      Displaying user list
    </h4>
  </div>
  <div
    className="container card-container"
  >
    <div
      className="row"
    >
      <UserCard
        key="1"
        markUserToRemove={[MockFunction]}
        removeUser={[MockFunction]}
        user={
          Object {
            "id": 1,
            "username": "astro",
          }
        }
      />
      <UserCard
        key="2"
        markUserToRemove={[MockFunction]}
        removeUser={[MockFunction]}
        user={
          Object {
            "id": 2,
            "username": "chemical",
          }
        }
      />
      <UserCard
        key="3"
        markUserToRemove={[MockFunction]}
        removeUser={[MockFunction]}
        user={
          Object {
            "id": 3,
            "username": "physics",
          }
        }
      />
    </div>
  </div>
</div>
`;
